src/imageResource.js

```diff
@@ -0,0 +1,177 @@
+'use strict';
+
+require('openseadragon');
+
+var ImageResource = function(config, parent, dispatcher) {
+  this.needed = config.needed || false;
+  this.visible = config.visible || false;
+  this.clipRegion = config.clipRegion;
+  this.opacity = config.opacity || 1;
+  this.x = config.x || 0;
+  this.y = config.y || 0;
+  this.height = config.height || 1;
+  this.width = config.width || 1;
+  this.zIndex = config.zIndex || 0;
+  this.tileSource = config.tileSource;
+  this.dynamic = config.dynamic || false;
+  this.imageType = config.imageType || "main"; // can be 'main', 'alternate', or 'detail'
+  this.status = 'initialized'; // can be 'requested', 'received', 'pending','shown', or 'failed'
+  this.parent = parent;
+  this.dispatcher = dispatcher;
+};
+
+ImageResource.prototype = {
+  hide: function(immediately) {
+    this.previousOpacity = this.opacity;
+    this.setOpacity(0, true);
+    this.visible = false;
+  },
+
+  show: function(immediately) {
+    this.visible = true;
+    if(this.previousOpacity) {
+      this.setOpacity(this.previousOpacity, true);
+    } else {
+      this.setOpacity(1, true);
+    }
+  },
+
+  // todo: take parent opacity into account here and in the constructor
+  setOpacity: function(opacity, immediately) {
+    this.opacity = opacity;
+    if(this.tiledImage) {
+      this.tiledImage.setOpacity(this.opacity, immediately);
```

Comment:
It would be totally sweet if OSD had fading for setOpacity, but it doesn't, so the `immediately` here doesn't actually do anything (it's always immediate).